### PR TITLE
Ft export workouts 15711085

### PR DESCRIPTION
**Pull request by frankip**
#### What does this PR do?
It implements the importing and exporting of workouts
#### Description of Task to be completed?
a user should be able to import and export workout and exerscises
#### How should this be manually tested?
after cloning and setting up the repo locally, change the branch to `ft-export-workouts-15711085` navigate to workout overview and add workout then add and training day and exercises. click on the options button you will see a export workout 
screen shot 2018-05-11 at 11 21 57

it will then download a csv file to your download folder

screen shot 2018-05-11 at 07 39 08

to import a csv navigate to the workout overview and adjacent to the add  button there is an import button. 
screen shot 2018-05-11 at 11 29 45

screen shot 2018-05-11 at 07 40 52

screen shot 2018-05-11 at 07 41 14

### CODE REVIEW

**Review by betsybeth — CHANGES_REQUESTED** [review]
Kindly fix conflicting files

File: wger/manager/views/workout.py [hunk 1]
```diff
@@ -406,3 +419,90 @@ def timer(request, day_pk):
...
+                day.day.add(DaysOfWeek.objects.filter(day_of_week=day_name).first())
+
+            single_set = Set(exerciseday=day)
+            single_set.save()
+            
+            # save the exercises 
+            for exercise in single_workout["Exercise"].split("\n"):
+                single_set.exercises.add(Exercise.objects.filter(name=exercise).first())
+
+    return HttpResponseRedirect(reverse('manager:workout:overview'))
```
Comment (Kalela) [inline wger/manager/views/workout.py]: Great job with your task. Quick question, for my own knowledge, are there other ways of exporting and importing json data apart from csv files?

**Review by Kalela — APPROVED** [review]

File: wger/manager/views/workout.py [hunk 2]
```diff
@@ -406,3 +414,90 @@ def timer(request, day_pk):
...
+    ''' implements the export workout as csv'''
+
+    workouts = Workout.objects.filter(user=request.user, id=pk)
+
+    # assign correct type and name to the file
+    response = HttpResponse(content_type='text/csv')
+    response['Content-Disposition'] = 'attachment; filename=' + str(request.user) + '_workouts.csv'
+    writer = csv.writer(response)
+    # populate the csv with the row names
+    writer.writerow(['Date created', 'Comment', 'Days', 'Description', 'Exercise'])
```
Comment (meshnesh) [inline wger/manager/views/workout.py:431]: Nice implementation of csv and export functionality

**Review by meshnesh — APPROVED** [review]

File: wger/manager/templates/workout/overview.html [hunk 3]
```diff
@@ -33,4 +33,43 @@ <h4 class="list-group-item-heading">{{ workout }}</h4>
...
+        <div class="modal-content">
+            
+            <div class="modal-header">
+                
+                <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+                
+                <h4 class="modal-title">{% trans "Import Workout" %}</h4>
+                </div>
+            
+            <div class="modal-body">
```
Comment (betsybeth) [inline wger/manager/templates/workout/overview.html:50]: Nice work on following the flake8 conventions eg identation

File: wger/manager/views/workout.py [hunk 4]
```diff
@@ -406,3 +414,90 @@ def timer(request, day_pk):
...
+
+    for workout in workouts:
+        # extract the workout days
```
Comment (betsybeth) [inline wger/manager/views/workout.py:434]: LGTM 
Good work commenting

**Review by betsybeth — APPROVED** [review]

**Review by mutoromike — APPROVED** [review]
LGTM

File: wger/manager/views/workout.py (hunk 1 above)
Comment (frankip) [reply wger/manager/views/workout.py]: CSV files are the international standard, from there its possible to convert them to any format

**Comment by lakivisi-zz** [comment]
@FelixWambiri @frankip waiting for your approvals

**Comment by lakivisi-zz** [comment]
@FelixWambiri Awaiting your approval?

**Review by FelixWambiri — APPROVED** [review]
LGTM.Nice work on the comments

**Pull request merged**